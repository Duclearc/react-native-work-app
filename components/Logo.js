import { Image } from "react-native"
import { StyleSheet } from "react-native";

export const styles = {
    container: {
      paddingTop: 50,
    },
    s: {
      width: 50,
      height: 50,
    },
    m: {
      width: 66,
      height: 58,
    },
    xl: {
      minHeight: 58
    }
  };

export const Logo = ({size}) => {
    size = size ? size:'s';

    return (
        <Image source={require('../assets/icon.png')} style={StyleSheet.create(styles[size])} />
    )
}