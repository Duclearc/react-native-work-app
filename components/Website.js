import { Platform, StyleSheet } from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { WebView } from 'react-native-webview';
import Config from "react-native-config";

export default function Website(props) {
    const url = props.url || Config.WEBSITE_URL;


    const styles = StyleSheet.create({
        ...Platform.select({
            ios: {
                marginTop: 30
            },
            android: {
                marginBottom: 50
            }
        })
    });

    return (
        <SafeAreaProvider>
            <WebView source={{ uri: url }} style={styles} />
        </SafeAreaProvider>
    );
};