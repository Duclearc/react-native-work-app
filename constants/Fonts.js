import { StyleSheet } from "react-native";
const defaultFont = "OpenSans_300Light";
const boldFont = "OpenSans_700Bold";
const rem = (fontSize, increaseForMobile = 2) => 0.625 * 16 * fontSize * increaseForMobile;

export const fontStyles = {
  lemFont: {
      fontFamily: defaultFont,
      color: "#333333",
      fontSize: rem(1.1)
    },
    lemLink: {
      fontFamily: defaultFont,
      color: "#073d6b"
    },
    lemBold: {
      fontFamily: boldFont,
      color: "#2a2c34"
    },
    lemHeading: {
      lineHeight: rem(2.5),
      fontSize: rem(2)
    }
}
export default StyleSheet.create(fontStyles);