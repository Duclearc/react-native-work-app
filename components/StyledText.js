import { Text } from 'react-native';
import Fonts from '../constants/Fonts';

export function StyledText(props) {
  return <Text {...props} style={[Fonts.lemFont, props.style]} />;
}
