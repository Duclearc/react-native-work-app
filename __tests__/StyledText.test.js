import React from 'react';
import { create } from 'react-test-renderer';
import { StyledText } from '../components/StyledText';
import { fontStyles } from '../constants/Fonts';

describe('<StyledText />', () => {
  jest.useFakeTimers();
  const screen = create(<StyledText />).toJSON();

  it('renders correctly', () => {
    expect(screen).toMatchSnapshot();
  });

  it(`loads stylesheet`, () => {
    expect(screen.props.style[0]).toBe(fontStyles.lemFont);
  });
});