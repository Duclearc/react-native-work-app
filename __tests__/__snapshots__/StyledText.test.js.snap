// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledText /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontFamily": "OpenSans_300Light",
        "fontSize": 22,
      },
      undefined,
    ]
  }
/>
`;
